pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -466,19 +466,21 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloud.MachineCreateDe
 	}
 	awsInstance := &awsInstance{instance: runOut.Instances[0]}
 
-	// Change to our security group
-	_, err = ec2Client.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
-		InstanceId: runOut.Instances[0].InstanceId,
-		Groups:     aws.StringSlice(config.SecurityGroupIDs),
-	})
-	if err != nil {
-		_, err := ec2Client.TerminateInstances(&ec2.TerminateInstancesInput{
-			InstanceIds: []*string{runOut.Instances[0].InstanceId},
+	if len(config.SecurityGroupIDs) > 0 {
+		// Change to our security group
+		_, modifyInstanceErr := ec2Client.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
```

Comment:
previously `err` was hidden and not returned to a developer, see line 476

Reply:
The question I have here thought is if there is any situation were not having a security group makes sense. It means the instance wont accept any incoming traffic which means it is basically useless as a kubernetes node. Maybe we should instead add a check in `Validate` that there is at least one security group defined?

Or are there other ways in which sec groups can be attached to instances @mrIncompetent ?

Reply:
@mrIncompetent can you please answer the question :) ?

Reply:
Adding a validation step:
```go
if len(config.SecurityGroupIDs) > 0 {
    return fmt.Errorf('no security groups specified')
}
```
should be the way to go

Reply:
alright, so having at least one security group is required. I'll update the code.